evennia/web/api/README.md

```diff
@@ -0,0 +1,102 @@
+
+Having an API can allow you to have richer interactions with client applications. For
+example, suppose you want to allow players to send and receive in-game messages from
+outside the game. You might define an endpoint that will retrieve all of a character's
+messages and returns it as a JSON response. Then in a webpage, you have a button that
+the user can click to make an [AJAX][ajax] request to that endpoint, retrieves the data, and
+displays it on the page. You also provide a form to let them send messages, where the
+submit button uses AJAX to make a POST request to that endpoint, sending along the
+JSON data from the form, and then returns the response of the results. This works,
+but then a tech-savvy player might ask if they can have their own application that
+will retrieve messages periodically for their own computer. By having a [REST][rest] API that
+they can use, they can create client applications of their own to retrieve or change
+data.
+
+Other examples of what you might use a RESTful API for would be players managing
+tasks out-of-game like crafting, guild management, retrieving stats on their
+characters, building rooms/grids, editing character details, etc. Any task that
+doesn't require real-time 2-way interaction is a good candidate for an API endpoint.
+
+## Sample requests
+
+The API contains a number of views already defined. If the API is enabled, by
+setting `REST_API_ENABLED = True` in your `settings.py`, endpoints will be
+accessible by users who make authenticated requests as users with builder
+permissions. Individual objects will check lockstrings to determine if the
+user has permission to perform retrieve/update/delete actions upon them.
+
+To view an object, you might make a request by using the 
+excellent [requests library][requests]:
+```pythonstub
+>>> import requests
+>>> response = requests.get("https://www.mygame.com/api/objects/57",
+                            auth=("Myusername", "password123"))
+>>> response.json()
+{"db_key": "A rusty longsword", "id": 57, "db_location": 213, ...}
+```
+The above example makes a GET request to the /objects/ endpoint to retrieve the
+object with an ID of 57, retrieving basic data for it. Now suppose that you want
+to use the API to create an object:
+```pythonstub
+>>> data = {"db_key": "A shiny sword"}
+>>> response = requests.post("https://www.mygame.com/api/objects",
+                             data=data, auth=("Anotherusername", "sekritpassword"))
+>>> response.json()
+{"db_key": "A shiny sword", "id": 214, "db_location": None, ...}
+```
+In the above example, you make a POST request to the /objects/ endpoint with
+the name of the object you wish to create passed along as data. Now suppose you
+decided you didn't like the name, and wanted to change it for the newly created
+object:
+```pythonstub
+>>> data = {"db_key": "An even SHINIER sword", "db_location": 50}
+>>> response = requests.post("https://www.mygame.com/api/objects/214",
```

Comment:
This will lead to a 405 `Method not allowed` error. This is a PATCH operation. Currently it also requires an ending slash.